plugin/pulpcore/plugin/changeset/iterator.py

```diff
@@ -166,13 +169,21 @@ def _iter(self):
         Yields:
             Download: The appropriate download object.
         """
-        for artifact in ArtifactIterator(self.content):
-            if self.deferred or artifact.model.downloaded:
+        for remote_artifact in ArtifactIterator(self.content):
+            q = Q()
+            for algorithm in hashlib.algorithms_guaranteed:
+                if getattr(remote_artifact.model, algorithm):
+                    kwargs = {algorithm: getattr(remote_artifact.model, algorithm)}
+                    q |= Q(**kwargs)
+            try:
+                artifact = Artifact.objects.get(q)
+            except Artifact.DoesNotExist:
+                artifact = None
+            if artifact or self.deferred:
                 download = NopDownload()
-                download.attachment = artifact
-                artifact.path = None
+                download.attachment = remote_artifact
             else:
-                download = artifact.download
```

Comment:
As mentioned earlier.  The convention in the package is that:

`content` is `RemoteContent`
and
`artifact` is `RemoteArtifact`

And that `model` was used for DB models.  If you want to change this convention, you need to do a mass rename.  Though it seems outside the scope of this work.